bundles/framework/announcements/view/AnnouncementsHandler.js

```diff
@@ -1,91 +1,72 @@
-import React from 'react';
-import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
-import { Message } from 'oskari-ui';
-import { LOCAL_STORAGE_KEY, LOCAL_STORAGE_SEPARATOR } from './Constants';
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+import { isUpcoming, isOutdated, isActive } from '../service/util';
 
 // Handler for announcements. Handles state and service calls.
-
-const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='announcements' />;
-
 class ViewHandler extends StateHandler {
     constructor (service) {
         super();
         this.service = service;
-        this.state = {
-            dontShowAgain: [],
-            announcements: [],
-            showAsPopup: []
-        };
-        this.initState();
+        this.service.on('tool', () => this.updateState({ tools: this.service.getTools() }));
+        this.service.on('fetch', () => this.onFetch());
+        this.service.fetchAnnouncements();
     }
 
-    initState () {
-        this.service.fetchAnnouncements(function (err, announcements) {
-            if (err) {
-                Messaging.error(getMessage('messages.getFailed'));
-            } else {
-                const dontShowAgain = this.getDontShowAgainIds(announcements);
-                const showAsPopup = announcements.filter(ann => ann.options.showAsPopup && !dontShowAgain.includes(ann.id));
-                this.updateState({
-                    announcements,
-                    dontShowAgain,
-                    showAsPopup
-                });
-            }
-        }.bind(this));
-    }
+    onFetch () {
+        const allAnnouncements = this.service.getAnnouncements();
+        const dontShowAgain = this.service.getIdsFromLocalStorage();
+
+        // Admin gets all announcements
+        const announcements = allAnnouncements.filter(a => isActive(a));
```

Comment:
I would rename some variables here to make it more clear for reading but it's no biggie (I had to scroll up to see if announcements was everything or the active ones :)):
- announcements -> activeAnnouncements
- allAnnouncements -> announcements

Reply:
announcements -> active (like outdated and upcoming)